### ISSUE TITLE
Make gzip/deflate compression configurable by the user

### ISSUE THREAD

**Issue by MigrantP** [restsharp/RestSharp#561]
In Http.Async.cs and Http.Sync.cs, it hardcodes the AutomaticDecompression property of the web request so that deflate & gzip are always requested. In certain environments, this causes a problem because the decompressors are not actually available. Specifically, when using Unity (mono) on Android and iOS devices, there's an exception when the compressed response comes back (see below).

If I change the lines to have only DecompressionMethods.None, it works great. It would be useful to be able to control this setting when using RestSharp and not have it be hardcoded.

iOS:

```
ExecutionEngineException: Attempting to JIT compile method '(wrapper native-to-managed) System.IO.Compression.DeflateStream:UnmanagedRead (intptr,int,intptr)' while running with --aot-only.

  at (wrapper managed-to-native) System.IO.Compression.DeflateStream:CreateZStream (System.IO.Compression.CompressionMode,bool,System.IO.Compression.DeflateStream/UnmanagedReadOrWrite,intptr)
  at System.IO.Compression.DeflateStream..ctor (System.IO.Stream compressedStream, CompressionMode mode, Boolean leaveOpen, Boolean gzip) [0x00000] in <filename unknown>:0 
  at System.IO.Compression.GZipStream..ctor (System.IO.Stream compressedStream, CompressionMode mode, Boolean leaveOpen) [0x00000] in <filename unknown>:0 
  at System.IO.Compression.GZipStream..ctor (System.IO.Stream compressedStream, CompressionMode mode) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebResponse..ctor (System.Uri uri, System.String method, System.Net.WebConnectionData data, System.Net.CookieContainer container) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebRequest.SetResponseData (System.Net.WebConnectionData data) [0x00000] in <filename unknown>:0 
Rethrow as WebException: Attempting to JIT compile method '(wrapper native-to-managed) System.IO.Compression.DeflateStream:UnmanagedRead (intptr,int,intptr)' while running with --aot-only.

  at System.Net.HttpWebRequest.EndGetResponse (IAsyncResult asyncResult) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebRequest.GetResponse () [0x00000] in <filename unknown>:0 
  at RestSharp.Http.GetRawResponse (System.Net.HttpWebRequest request) [0x00000] in <filename unknown>:0 
```

Android:

```
System.EntryPointNotFoundException : CreateZStream---EXCEPTION---at (wrapper managed-to-native) System.IO.Compression.DeflateStream:CreateZStream (System.IO.Compression.CompressionMode,bool,System.IO.Compression.DeflateStream/UnmanagedReadOrWrite,intptr)at System.IO.Compression.DeflateStream..ctor (System.IO.Stream compressedStream, CompressionMode mode, Boolean leaveOpen, Boolean gzip) [0x00000] in :0at System.IO.Compression.DeflateStream..ctor (System.IO.Stream compressedStream, CompressionMode mode) [0x00000] in :0at (wrapper remoting-invoke-with-check) System.IO.Compression.DeflateStream:.ctor (System.IO.Stream,System.IO.Compression.CompressionMode)
```

**Comment by marc-at-nowhow** [comment]
Yes, I would be **most happy** to see this feature added, too! I'm facing a case where I really need to be able to configuratively **TURN OFF** the GZIP support....

**Comment by JustinY17** [comment]
I also need to turn this off.

**Comment by Fabman08** [comment]
I also need this feature!

**Comment by amantur** [comment]
Yes ,please need the ability to turn headers off or clear them altogether.

**Comment by OtiliaZaharia** [comment]
We badly need this feature too! It would be great is a way to change default header values.
If not, we'll sadly have to move to a different API client as we use this to automate our API tests

**Comment by Fabman08** [comment]
Hi guys, yesterday I published a new fork with the fix.
Take a look to https://github.com/Fabman08/RestSharp/commit/85c20480bc08b17d6db6d8f30df84984b2d4aaeb

**Comment by SuperShowwei** [comment]
I need it too.
It happened exception that i got the response from Autonomy Search Engine request with deflate method.

I need set no compression on request.

**Comment by pbudiardjo** [comment]
I need it too please

**Comment by jchannon** [comment]
@Fabman08 did you send a PR for your fork?

**Comment by jchannon** [comment]
Sent a PR from your commit

**Comment by Fabman08** [comment]
@jchannon thank you!

**Comment by jchannon** [comment]
no worries, lets hope it can be pulled asap 😄  @hallem

**Comment by kahalm** [comment]
We would need that feature as well.

**Comment by alexeyzimarev** [comment]
Fixed by #832
